src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -697,18 +697,48 @@ private static int calculateTagColumn(Matcher javadocTagMatcher,
             final List<DetailAST> throwLiterals = findTokensInAstByType(blockAst,
                     TokenTypes.LITERAL_THROW);
             for (DetailAST throwAst : throwLiterals) {
-                final DetailAST newAst = throwAst.getFirstChild().getFirstChild();
-                if (newAst.getType() == TokenTypes.LITERAL_NEW) {
-                    final FullIdent ident = FullIdent.createFullIdent(newAst.getFirstChild());
-                    final ExceptionInfo exceptionInfo = new ExceptionInfo(
-                            createClassInfo(new Token(ident), currentClassName));
-                    returnValue.add(exceptionInfo);
+                if (!isInTryCatchOrLambda(blockAst, throwAst)) {
+                    final DetailAST newAst = throwAst.getFirstChild().getFirstChild();
+                    if (newAst.getType() == TokenTypes.LITERAL_NEW) {
+                        final FullIdent ident = FullIdent.createFullIdent(newAst.getFirstChild());
+                        final ExceptionInfo exceptionInfo = new ExceptionInfo(
+                                createClassInfo(new Token(ident), currentClassName));
+                        returnValue.add(exceptionInfo);
+                    }
                 }
             }
         }
         return returnValue;
     }
 
+    /**
+     * Checks if a 'throw' usage is contained within a try block or lambda expression.
+     * Note that a try block without catch is not considered.
```

Comment:
We probably should make example in xdoc to explain user that ignore lambdas at all because we do not know for sure when it will be evaluated (in method or later on smoewhere else) as it is kind-of-anonymous-class with method, so it should have its own javadoc (but nobody do javadocs for lambda .... so we ignore it).

Reply:
Added some notes in the check description:
![image](https://user-images.githubusercontent.com/53135010/79689461-8a420600-8287-11ea-8494-2fc1566eb29c.png)

The `ATTENTION:` part was originally from the examples below, I moved it up here so it's all in one place.

Added examples:
![image](https://user-images.githubusercontent.com/53135010/79689471-a47be400-8287-11ea-9661-1fbf5a349f94.png)
![image](https://user-images.githubusercontent.com/53135010/79689475-b493c380-8287-11ea-839d-77744c152671.png)

Reply:
Sample run for the new examples:
```
D:\checkstyletest>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <module name="TreeWalker">
        <module name="JavadocMethod">
            <property name="validateThrows" value="true"/>
        </module>

    </module>
</module>

D:\checkstyletest>type test\Test.java
import org.mockito.internal.matchers.Null;

class Test {

    /**
     * Ignore try block, but keep catch and finally blocks.
     *
     * @param s String to parse
     * @return A positive integer
     */
    public int parsePositiveInt(String s) {
        try {
            int value = Integer.parseInt(s);
            if (value <= 0) {
                throw new NumberFormatException(value + " is negative/zero"); // ok, try
            }
            return value;
        } catch (NumberFormatException ex) {
            throw new IllegalArgumentException("Invalid number", ex); // violation, catch
        } finally {
            throw new IllegalStateException("Should never reach here"); // violation, finally
        }
    }

    /**
     * Try block without catch is not ignored.
     *
     * @return a String from standard input, if there is one
     */
    public String readLine() {
        try (Scanner sc = new Scanner(System.in)) {
            if (!sc.hasNext()) {
                throw new IllegalStateException("Empty input"); // violation, not caught
            }
            return sc.next();
        }
    }

    /**
     * Lambda expressions are ignored as we do not know when the exception will be thrown.
     *
     * @param s a String to be printed at some point in the future
     * @return a Runnable to be executed when the string is to be printed
     */
    public Runnable printLater(String s) {
        return () -> {
            if (s == null) {
                throw new NullPointerException(); // ok
            }
            System.out.println(s);
        };
    }

}

D:\checkstyletest>java -jar -Duser.language=en -Duser.country=US checkstyle-8.32-SNAPSHOT-all.jar -c config.xml test\Test.java
Starting audit...
[ERROR] D:\checkstyletest\test\Test.java:19:23: Expected @throws tag for 'IllegalArgumentException'. [JavadocMethod]
[ERROR] D:\checkstyletest\test\Test.java:21:23: Expected @throws tag for 'IllegalStateException'. [JavadocMethod]
[ERROR] D:\checkstyletest\test\Test.java:33:27: Expected @throws tag for 'IllegalStateException'. [JavadocMethod]
Audit done.
Checkstyle ends with 3 errors.
```